Bottom: b8f38bd395b8c01c31a95b3a712bb38078840d1e
Top:    b9ba4dcdb960d5ea5aeca617392289011f30e9bb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 16:43:36 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5127430..2d65872 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1161,7 +1161,11 @@ label_contour (PLINE * a)
 	  continue;
 	}
     }
-  while ((cur = cur->next) != first_not_unknown);
+  while ((cur = cur->next) != first_not_unknown &&
+	 !(cur->next == &a->head && first_not_unknown == NULL));
+
+  if (first_not_unknown == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
